e2e/test/iothub/telemetry/TelemetrySendE2ETests.cs

```diff
@@ -0,0 +1,261 @@
+
+            await deviceClient.OpenAsync().ConfigureAwait(false);
+            await SendSingleMessageWithComponentAsync(deviceClient, testDevice.Id, Logger, messageSize).ConfigureAwait(false);
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+        }
+
+        private async Task SendTelemetryWithComponent(TestDeviceType type, ITransportSettings[] transportSettings, int messageSize = 0)
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, DevicePrefix, type).ConfigureAwait(false);
+            using DeviceClient deviceClient = testDevice.CreateDeviceClient(transportSettings);
+
+            await deviceClient.OpenAsync().ConfigureAwait(false);
+            await SendSingleMessageWithComponentAsync(deviceClient, testDevice.Id, Logger, messageSize).ConfigureAwait(false);
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+        }
+
+        public static async Task SendSingleMessageAsync(DeviceClient deviceClient, string deviceId, MsTestLogger logger, int messageSize = 0)
+        {
+            Client.TelemetryMessage testMessage;
+
+            if (messageSize == 0)
+            {
+                (testMessage, _) = ComposeTelemetryMessage(logger);
+            }
+            else
+            {
+                (testMessage, _) = ComposeComplexTelemetryMessage(messageSize, logger);
+            }
+
+            using (testMessage)
+            {
+                await deviceClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
+            }
+        }
+
+        public static async Task SendSingleMessageWithComponentAsync(DeviceClient deviceClient, string deviceId, MsTestLogger logger, int messageSize = 0)
+        {
+            Client.TelemetryMessage testMessage;
+
+            if (messageSize == 0)
+            {
+                (testMessage, _) = ComposeTelemetryMessageWithComponent(logger);
+            }
+            else
+            {
+                (testMessage, _) = ComposeComplexTelemetryMessageWithComponent(messageSize, logger);
+            }
+
+            using (testMessage)
+            {
+                await deviceClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
+            }
+        }
+
+        public static (Client.TelemetryMessage message, string p1Value) ComposeTelemetryMessageWithComponent(MsTestLogger logger)
+        {
+            string messageId = Guid.NewGuid().ToString();
+            string p1Value = Guid.NewGuid().ToString();
+            string userId = Guid.NewGuid().ToString();
+            string componentName = Guid.NewGuid().ToString();
+
+            logger.Trace($"{nameof(ComposeTelemetryMessage)}: messageId='{messageId}' userId='{userId}' p1Value='{p1Value}'");
+            var message = new TelemetryMessage
+            {
+                MessageId = messageId,
+                UserId = userId,
+                ComponentName = componentName,
```

Comment:
we should maybe add a unit test that checks that setting the component name via TelemetryCollection constructor works as well?